experimental/adaptive-tool/README.md

```diff
@@ -1,57 +1,122 @@
+![template_definition](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/main/experimental/adaptive-tool/resources/images/template_definition.gif)
 
-- diagnostic check
-![lg_diagnostic](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/lg_diagnostic.gif)
+## Debugging
+Adaptive Tool lets developers debug .dialog and LG files in runtime.
 
-- template reference hover
-![template_hover](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/template_hover.png)
+### Configuration
+- Install the adaptive-tool extension.
+- Open the LG or .dialog file  to debug.
+- To configure Visual Studio Code, add a target in your `launch.settings` file.
 
-- builtin function hover
-![function_hover](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/function_hover.png)
+This is an example of a typical `launch.json`:
+```json
+{
+    "type": "json",
+    "request": "attach",
+    "name": "Attach to Dialog",
+    "debugServer": 4712
+}
+```
 
-- buildin function and template suggestion
-![function_template_suggestion](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/function_template_suggestion.gif)
+`debugServer` refers to the port bot runs on.
 
-- structure property suggestion
-![structure_suggestion](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/structure_suggestion.gif)
+### Start bot runtime and complete initialization
 
-- template definition
-![template_definition](https://raw.githubusercontent.com/microsoft/BotBuilder-Samples/master/experimental/adaptive-tool/resources/images/template_definition.gif)
+Start a bot runtime. For example, start a bot project: `todobot` in [SampleBots](https://github.com/microsoft/botbuilder-dotnet/tree/hond/debugger/tests/Microsoft.Bot.Builder.TestBot.Json). Then make sure the debugger port has been registered in `BotFrameworkHttpAdapter` with `UseDebugger` method.
 
-## debugging feature
- see more details in [debugging](./debugging.md)
+There are several ways to initialize a bot, and [BotFremawork Emulator](https://github.com/microsoft/BotFramework-Emulator) is a typical approach. Open the emulator and attach it to the bot to finish the initialization.
 
+### Debug the runtime
+- Run the Visual Studio Code program by clicking **F5** and setting break points in the .dialog or LG file.
+- Chat with the bot in the emulator.
+- The cursor will stop when the code hits any of corresponding breakpoints.
 
-# how to build and use this extension
-- `npm install`
-- `npm run build`
+
+# Useful Commands
+|Command|Description|
+|-----|---------------|
+|`lgLiveTest.start`|Start a WebView to test the current LG file|
+
+
+# Adaptive-tool Settings
+|Name|Description|
+|-----|---------------|
```

Comment:
Add an intro on how this lets you ignore or add custom function recognition.

Reply:
Solved